src/virtualElements.js

```diff
@@ -133,19 +140,30 @@
         },
 
         firstChild: function(node) {
-            if (!isStartComment(node))
+            if (!isStartComment(node)) {
+                if (node.firstChild && isUnmatchedEndComment(node.firstChild)) {
+                    throw new Error("Found end comment without opening comment, as first child of " + node);
+                }
                 return node.firstChild;
-            if (!node.nextSibling || isEndComment(node.nextSibling))
```

Comment:
I don't think this should be removed. If you have `<!-- ko text: abc --><!-- /ko -->`, and call `firstChild`, it should return `null`, but the new code will return the closing comment.

Reply:
True. Updated.